compiler/luci/export/src/CircleTensorExporter.cpp

```diff
@@ -247,12 +247,21 @@ flatbuffers::Offset<Vector<int32_t>> encodeShape(FlatBufferBuilder &builder,
 }
 
 flatbuffers::Offset<Vector<int32_t>> encodeShapeSignature(FlatBufferBuilder &builder,
-                                                          const ShapeSignature &shape_signature)
+                                                          const ShapeDescription &shape)
 {
-  if (shape_signature.rank() == 0)
-    return 0;
+  assert(shape._rank_known && "unknown number of dimensions is not supported");
```

Comment:
does the caller ensure this condition?
or do we have some test models with unknown rank so that the caller do something?

Reply:
The only case that `shape._rank_known` is `false` is when `ShapeStatus` of the node is `UNDEFINED` or `NOSHAPE`.

https://github.com/Samsung/ONE/blob/1733eec56afb19d045cd775ab89f21b4feb3340e/compiler/luci/export/src/CircleTensorExporter.cpp#L116-L117

These case are kept when `encodeShapeSignature` is called.

https://github.com/Samsung/ONE/blob/e60ddf7524ddf374632f71d9c59aabca3a6d5ae4/compiler/luci/export/src/CircleTensorExporter.cpp#L457-L461

This feature is already used in `encodeShape`.
I think that https://github.com/Samsung/ONE/issues/5510#issuecomment-752337908 may be the reason :D

Reply:
Thanks for the explanation, but isn't there possibility that rank maybe unknown? I was wondering for that case.

Reply:
I think we can go as is. question was just from curiosity.

Reply:
Currently, it is impossible that some models have unknown rank.
For now, if shape inference impossible, just [] is returned, which is rank-0.